### PR TITLE
Ports plasma glass windows from /vg/

### PR DESCRIPTION
**Pull request by ghost**
:cl: somebody
add: Strong plasma glass windows
/:cl:

because WHY would ANYONE protect, say, the freaking SHARD with a window filled with rods? We need plasma windows instead!

i have no idea if this will work, i did this in the online 

- [x] Make window textures/smoothing
- [x] Make sheet crafting
- [x] Make sheet smelting
- [x] Finish sheet textures

reinforced on the left, normal on the right

### CODE REVIEW

**Comment by WJohn** [comment]
i feel like these have been ported about a million times.

**Comment by MoreRobustThanYou** [comment]
I looked at that... and apparently /vg/ doesn't have smoothing textures and I DON'T KNOW HOW TO SMOOTHIE TEXTURES

**Comment by juanototo2** [comment]
I'll take it. Basically window rwalls that aren't piss easy to get through... Sure, sure.

**Comment by MoreRobustThanYou** [comment]
I would make reinforced plasma glass an alloy, but I don't know how (you get it from ORM rather that smacking glass with plasma). Also, they're still very breakable by bullets.

**Comment by WJohn** [comment]
I mean we should replace high pressure windows with these, because they effectively fulfill that exact purpose. Would take about 10 seconds to do per map.

**Comment by MoreRobustThanYou** [comment]
How do I make smoothing textures...?

**Comment by WJohn** [comment]
For the icon process you can mimic the many existing smooth turf/table files. Check out the abductor tables since they give you the best idea of what each corner is supposed to be, then try and replace those 16x16 pixel corners with the equivalent one from plasma windows. You don't have to rename anything else in there (except maybe the full sprite used for the map maker).

**Comment by MoreRobustThanYou** [comment]
so basically copy/pasting in paint net?

**Comment by MoreRobustThanYou** [comment]
ugh.. I could make the corners.. but the sides confused me.

**Comment by MoreRobustThanYou** [comment]
if the smoothing icons managed to get made, and this merged, would you guys like to see plasma/reinforced plasma glass in places like the SME, bridge, other important places?

**Comment by WJohn** [comment]
Well they should definitely replace wherever high-pressure windows are currently being used, a simple find-path-replace-all would handle it. As for usage on the bridge, armory, or other places, leave it up to people who want to do wade knee-deep in the balance meme shitstorm.

**Comment by DoctoreMad** [comment]
>I mean we should replace high pressure windows with these, because they effectively fulfill that exact purpose. 

That and it actually can be created.
~While we're at it, someone should add a way to reinforce windows against the screwdriver/crowbar/screwdriver deconstruction path which is way too fucking short. Plasma windows increase damage resistance but is useless for the armory.~ Oh wait this would be PvP balance.

**Comment by DoctoreMad** [comment]
>ugh.. I could make the corners.. but the sides confused me.

Its just normal glass recolored into purple

**Comment by FantasticFwoosh** [comment]
This was something i was going to do, but i never got round to it. Nice to see its being done.

**Comment by RemieRichards** [comment]
@MoreRobustThanYou just use the smoothing cutter.
https://github.com/tgstation/tgstation/tree/master/tools/SS13SmoothingCutter
You'll need to use the 'ExampleInput.dmi' template for windows.

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 1]
```diff
@@ -36,37 +36,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+			var/obj/item/stack/sheet/glass/plasma/G = src
 			src = null
 			var/replace = (user.get_inactive_held_item()==G)
 			V.use(1)
 			G.use(1)
 			if (!G && replace)
 				user.put_in_hands(RG)
 		else
-			to_chat(user, "<span class='warning'>You need one rod and one sheet of glass to make reinforced glass!</span>")
+			to_chat(user, "<span class='warning'>You need one rod and one sheet of plamsa glass to make reinforced plasma glass!</span>")
```
Comment (GunHog) [inline code/game/objects/items/stacks/sheets/glass.dm]: `plamsa`

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 2]
```diff
@@ -109,6 +128,26 @@ GLOBAL_LIST_INIT(reinforced_glass_recipes, list ( \
...
 	..()
+	
+GLOBAL_LIST_INIT(prglass_recipes, list ( \
+	new/datum/stack_recipe("directional reinforced window", /obj/structure/window/plasma/reinforced/unanchored, time = 0, on_floor = TRUE, window_checks = TRUE), \
+	new/datum/stack_recipe("fulltile reinforced window", /obj/structure/window/plasma/reinforced/fulltile/unanchored, 2, time = 0, on_floor = TRUE, window_checks = TRUE) \
+))
+
+/obj/item/stack/sheet/rglass/plasma
+	name = "reinforced plasma glass"
+	desc = "Plasma glass which seems to have rods or something stuck in them."
```
Comment (GunHog) [inline code/game/objects/items/stacks/sheets/glass.dm]: This needs a better desc. Perhaps "A plasma-glass alloy fortified with metal rods.", or such?

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 3]
```diff
@@ -109,6 +128,26 @@ GLOBAL_LIST_INIT(reinforced_glass_recipes, list ( \
...
+	singular_name = "reinforced plasma glass sheet"
+	icon_state = "sheet-plasmarglass"
+	materials = list(MAT_METAL=MINERAL_MATERIAL_AMOUNT/2, MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
```
Comment (GunHog) [inline code/game/objects/items/stacks/sheets/glass.dm]: No plasma materials?

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 2 above)
Comment (RemieRichards) [inline code/game/objects/items/stacks/sheets/glass.dm]: its the same as the normal reinforced glass.

File: code/game/objects/structures/window.dm [hunk 4]
```diff
@@ -426,6 +426,34 @@
...
+	max_integrity = 50
+	explosion_block = 1
+	glass_type = /obj/item/stack/sheet/glass/plasma
+
+/obj/structure/window/plasma/unanchored
+	anchored = FALSE
+
+/obj/structure/window/plasma/reinforced
+	name = "reinforced plasma window"
+	desc = "A window made out of a plasma-silicate alloy and a rod matrice. It looks hopelessly tough to break and is most likely nigh fireproof."
```
Comment (GunHog) [inline code/game/objects/structures/window.dm]: This is a wonderful desc, why not use something similar for the sheet item?

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 2 above)
Comment (GunHog) [inline code/game/objects/items/stacks/sheets/glass.dm]: Oh, that is just...sad.

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 3 above)
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: Oops!

File: code/game/objects/structures/window.dm (hunk 4 above)
Comment (MoreRobustThanYou) [inline code/game/objects/structures/window.dm]: it's copy/pasted from /vg/, but ok

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 5]
```diff
@@ -36,37 +36,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 75, acid = 100)
+	resistance_flags = ACID_PROOF
+	merge_type = /obj/item/stack/sheet/glass/plasma
+
+/obj/item/stack/sheet/glass/plasma/fifty
+	amount = 50
+
+/obj/item/stack/sheet/glass/plasma/Initialize(mapload, new_amount, merge = TRUE)
+	recipes = GLOB.pglass_recipes
+	..()
```
Comment (lzimann) [inline code/game/objects/items/stacks/sheets/glass.dm]: `return ..()`

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 6]
```diff
@@ -109,6 +128,26 @@ GLOBAL_LIST_INIT(reinforced_glass_recipes, list ( \
...
+	icon_state = "sheet-plasmarglass"
+	materials = list(MAT_METAL=MINERAL_MATERIAL_AMOUNT/2, MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
+	origin_tech = "materials=2;plasmatech=2"
+	armor = list(melee = 20, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 80, acid = 100)
+	resistance_flags = ACID_PROOF
+	merge_type = /obj/item/stack/sheet/rglass/plasma
+
+/obj/item/stack/sheet/rglass/Initialize(mapload, new_amount, merge = TRUE)
+	recipes = GLOB.prglass_recipes
+	..()
```
Comment (lzimann) [inline code/game/objects/items/stacks/sheets/glass.dm]: `return ..()`

File: code/game/objects/structures/window.dm [hunk 7]
```diff
@@ -426,6 +426,34 @@
...
+	icon_state = "plasmawindow"
+	reinf = 1
```
Comment (lzimann) [inline code/game/objects/structures/window.dm]: `TRUE`

File: code/game/objects/structures/window.dm [hunk 8]
```diff
@@ -426,6 +426,34 @@
 
 /obj/structure/window/reinforced/unanchored
 	anchored = FALSE
+	
+/obj/structure/window/plasma
+	name = "plasma window"
+	desc = "A window made out of a plasma-silicate alloy. It looks insanely tough to break and burn through."
+	icon_state = "plasmawindow"
+	reinf = 0
```
Comment (lzimann) [inline code/game/objects/structures/window.dm]: `FALSE`

Comment (MoreRobustThanYou) [inline code/game/objects/structures/window.dm]: the rest of the code has it like 0/1 I think.

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 9]
```diff
@@ -18,7 +18,7 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 	desc = "HOLY SHEET! That is a lot of glass."
 	singular_name = "glass sheet"
 	icon_state = "sheet-glass"
-	materials = list(MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
+	materials = list(MAT_PLASMA=MINERAL_MATERIAL_AMOUNT)
```
Comment (XDTM) [inline code/game/objects/items/stacks/sheets/glass.dm]: uh

Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: crap

**Comment by MoreRobustThanYou** [comment]
With the smoothing cutter... is  center_8 required? I can't find a file to match it

**Comment by MoreRobustThanYou** [comment]
got it... probably will look like crap because I did crappy paint.net trickery for center_8, but whatever

**Comment by MoreRobustThanYou** [comment]
this should work... but again I haven't slept one bit since I started this so the code is probably the quality of a dog's ass

**Comment by MoreRobustThanYou** [comment]
it compiles on travis ci... it works i thinkkf

**Comment by MoreRobustThanYou** [comment]
it compiles somhow

**Comment by MoreRobustThanYou** [comment]
it works. powered by magic

**Comment by MoreRobustThanYou** [comment]
@lzimann and @GunHog your changes have been implemented

**Comment by MoreRobustThanYou** [comment]
mmm how do you add smelter recipes?

**Comment by RemieRichards** [comment]
@MoreRobustThanYou for future reference, center_8 is what the object would look like if it was surrounded on all 8 sides. (center_4 being if it's surrounded NORTH,EAST,SOUTH and WEST)

**Comment by MoreRobustThanYou** [comment]
Ok.

**Comment by MoreRobustThanYou** [comment]
since the old, non-smoothing sprites didn't have that, i just did a crappy paint.net of it

**Comment by RemieRichards** [comment]
Ah, yours don't look quite right, compare your smoothing icons to window.dmi and you'll see your ones on the end (the -fs) aren't a match, I assume the same is true with your reinforced ones too.

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 10]
```diff
@@ -67,6 +82,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+))
+
+/obj/item/stack/sheet/glass/plasma
+	name = "plasmaglass"
+	desc = "A glass sheet made out of a plasma-silicate alloy. It looks extremely tough and heavily fire resistant."
+	singular_name = "plasma glass sheet"
+	icon_state = "sheet-plasmaglass"
+	materials = list(MAT_PLASMA=MINERAL_MATERIAL_AMOUNT/2, MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
+	origin_tech = "plasmatech=2;materials=2"
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 75, acid = 100)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: quote these associations

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 11]
```diff
@@ -89,6 +151,9 @@ GLOBAL_LIST_INIT(reinforced_glass_recipes, list ( \
 	resistance_flags = ACID_PROOF
 	merge_type = /obj/item/stack/sheet/rglass
 
+/obj/item/stack/sheet/rglass/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: call ..()

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 12]
```diff
@@ -108,7 +173,27 @@ GLOBAL_LIST_INIT(reinforced_glass_recipes, list ( \
...
+))
+
+/obj/item/stack/sheet/rglass/plasma
+	name = "reinforced plasma glass"
+	desc = "A glass sheet made out of a plasma-silicate alloy and a rod matrice. It looks hopelessly tough and nearly fire-proof!"
+	singular_name = "reinforced plasma glass sheet"
+	icon_state = "sheet-plasmarglass"
+	materials = list(MAT_PLASMA=MINERAL_MATERIAL_AMOUNT/2, MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
+	origin_tech = "materials=2;plasmatech=2"
+	armor = list(melee = 20, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 80, acid = 100)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: unquoted associations

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 13]
```diff
@@ -48,6 +48,21 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+		var/obj/item/stack/sheet/mineral/plasma/V = W
+		if (V.get_amount() >= 2 && src.get_amount() >= 1)
+			var/obj/item/stack/sheet/glass/plasma/RG = new (user.loc)
+			RG.add_fingerprint(user)
+			var/obj/item/stack/sheet/glass/G = src
+			src = null
+			var/replace = (user.get_inactive_held_item()==G)
+			V.use(1)
+			G.use(1)
+			if (!G && replace)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: if(QDELETED(G) && replace)

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 14]
```diff
@@ -48,6 +48,21 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
 		var/obj/item/stack/light_w/new_tile = new(user.loc)
 		new_tile.add_fingerprint(user)
+	else if(istype(W, /obj/item/stack/sheet/mineral/plasma))
+		var/obj/item/stack/sheet/mineral/plasma/V = W
+		if (V.get_amount() >= 2 && src.get_amount() >= 1)
+			var/obj/item/stack/sheet/glass/plasma/RG = new (user.loc)
+			RG.add_fingerprint(user)
+			var/obj/item/stack/sheet/glass/G = src
+			src = null
+			var/replace = (user.get_inactive_held_item()==G)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: no need for the brackets

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 15]
```diff
@@ -48,6 +48,21 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 		to_chat(user, "<span class='notice'>You attach wire to the [name].</span>")
 		var/obj/item/stack/light_w/new_tile = new(user.loc)
 		new_tile.add_fingerprint(user)
+	else if(istype(W, /obj/item/stack/sheet/mineral/plasma))
+		var/obj/item/stack/sheet/mineral/plasma/V = W
+		if (V.get_amount() >= 2 && src.get_amount() >= 1)
+			var/obj/item/stack/sheet/glass/plasma/RG = new (user.loc)
+			RG.add_fingerprint(user)
+			var/obj/item/stack/sheet/glass/G = src
+			src = null
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: no need for this

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 16]
```diff
@@ -48,6 +48,21 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 		to_chat(user, "<span class='notice'>You attach wire to the [name].</span>")
 		var/obj/item/stack/light_w/new_tile = new(user.loc)
 		new_tile.add_fingerprint(user)
+	else if(istype(W, /obj/item/stack/sheet/mineral/plasma))
+		var/obj/item/stack/sheet/mineral/plasma/V = W
+		if (V.get_amount() >= 2 && src.get_amount() >= 1)
+			var/obj/item/stack/sheet/glass/plasma/RG = new (user.loc)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: get_turf(user)

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 17]
```diff
@@ -48,6 +48,21 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 		to_chat(user, "<span class='notice'>You attach wire to the [name].</span>")
 		var/obj/item/stack/light_w/new_tile = new(user.loc)
 		new_tile.add_fingerprint(user)
+	else if(istype(W, /obj/item/stack/sheet/mineral/plasma))
+		var/obj/item/stack/sheet/mineral/plasma/V = W
+		if (V.get_amount() >= 2 && src.get_amount() >= 1)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: remove src

**Comment by MoreRobustThanYou** [comment]
I'm bad at this... i used the smoothing tool to do this

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 18]
```diff
@@ -67,6 +82,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+	add_fingerprint(user)
+
+	if(istype(W, /obj/item/stack/rods))
+		var/obj/item/stack/rods/V = W
+		if (V.get_amount() >= 1 && src.get_amount() >= 1)
+			var/obj/item/stack/sheet/rglass/plasma/RG = new (user.loc)
+			RG.add_fingerprint(user)
+			var/obj/item/stack/sheet/glass/plasma/G = src
+			src = null
+			var/replace = (user.get_inactive_held_item()==G)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: no braces

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 19]
```diff
@@ -67,6 +82,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+/obj/item/stack/sheet/glass/plasma/Initialize(mapload, new_amount, merge = TRUE)
+	recipes = GLOB.pglass_recipes
+	return ..()
+	
+/obj/item/stack/sheet/glass/plasma/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
+
+	if(istype(W, /obj/item/stack/rods))
+		var/obj/item/stack/rods/V = W
+		if (V.get_amount() >= 1 && src.get_amount() >= 1)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: remove src

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 20]
```diff
@@ -67,6 +82,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+			var/obj/item/stack/sheet/rglass/plasma/RG = new (user.loc)
+			RG.add_fingerprint(user)
+			var/obj/item/stack/sheet/glass/plasma/G = src
+			src = null
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: no need for this

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 21]
```diff
@@ -67,6 +82,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+	recipes = GLOB.pglass_recipes
+	return ..()
+	
+/obj/item/stack/sheet/glass/plasma/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
+
+	if(istype(W, /obj/item/stack/rods))
+		var/obj/item/stack/rods/V = W
+		if (V.get_amount() >= 1 && src.get_amount() >= 1)
+			var/obj/item/stack/sheet/rglass/plasma/RG = new (user.loc)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: get_turf(user)

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 22]
```diff
@@ -67,6 +82,53 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+			RG.add_fingerprint(user)
+			var/obj/item/stack/sheet/glass/plasma/G = src
+			src = null
+			var/replace = (user.get_inactive_held_item()==G)
+			V.use(1)
+			G.use(1)
+			if (!G && replace)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: if(QDELETED(G) && replace)

File: code/game/objects/structures/window.dm [hunk 23]
```diff
@@ -426,6 +426,34 @@
...
 /obj/structure/window/reinforced/unanchored
 	anchored = FALSE
+	
+/obj/structure/window/plasma
+	name = "plasma window"
+	desc = "A window made out of a plasma-silicate alloy. It looks insanely tough to break and burn through."
+	icon_state = "plasmawindow"
+	reinf = FALSE
+	heat_resistance = 2500
+	armor = list(melee = 75, bullet = 5, laser = 0, energy = 0, bomb = 45, bio = 100, rad = 100, fire = 90, acid = 100)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: unquoted associations

**Comment by RemieRichards** [comment]
I know you did, but the tool isn't wrong, what you gave it was, your center_8 looked like your center_4, where as for windows it would just be a solid (transparent, because windows) colour, take a look at window.dmi's -f states to see that they don't have the little corner shapes yours do.

File: code/game/objects/structures/window.dm [hunk 24]
```diff
@@ -426,6 +426,34 @@
...
+/obj/structure/window/plasma/unanchored
+	anchored = FALSE
+
+/obj/structure/window/plasma/reinforced
+	name = "reinforced plasma window"
+	desc = "A window made out of a plasma-silicate alloy and a rod matrice. It looks hopelessly tough to break and is most likely nigh fireproof."
+	icon_state = "rplasmawindow"
+	reinf = TRUE
+	heat_resistance = 4000
+	armor = list(melee = 85, bullet = 20, laser = 0, energy = 0, bomb = 60, bio = 100, rad = 100, fire = 99, acid = 100)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: unquoted associations

File: code/game/objects/structures/window.dm [hunk 25]
```diff
@@ -426,6 +426,34 @@
...
+	max_integrity = 100
+	explosion_block = 1.5
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: this is not a decimal value

**Comment by MoreRobustThanYou** [comment]
I honestly just copy-pasted the pattern onto the corners for the _8, but lemme look

**Comment by MoreRobustThanYou** [comment]
here's the reinforced plasma window file I used for the tool: https://ufile.io/b73tr (sorry, couldn't upload to github)

normal plasma: https://ufile.io/chqto

**Review by Cyberboss — CHANGES_REQUESTED** [review]
This PR depends on #29030

**Comment by RemieRichards** [comment]
Ah! your reinforced plasma center_8 is correct, but your normal plasma center_8 isn't.
The normal center_8 shouldn't have corners, just like the reinforced plasma center_8 doesn't.

**Comment by MoreRobustThanYou** [comment]
@RemieRichards would it just being fully pink (corner 8 that is) be good? since it's mostly pink anyways except for the large corners

**Comment by RemieRichards** [comment]
@MoreRobustThanYou Fully pink using the same transparent pink used in the rest of the states, yes.

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 10 above)
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: armor?

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 12 above)
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: armor? origin tech?

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 15 above)
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: all 3 lines or wat

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 12 above)
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: like do I do "melee" = 20?

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 20 above)
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/glass.dm]: wait, wouldn't the item not be consumed without this?

**Comment by MoreRobustThanYou** [comment]
@Cyberboss I would like to note the armor unquoted and stuff was like that in the original window.dm

**Comment by Cyberboss** [comment]
Just because it's old code doesn't mean it's not shitcode

**Comment by MoreRobustThanYou** [comment]
I'm guessing the armor needs quotes?

**Comment by Cyberboss** [comment]
Yes

**Comment by MoreRobustThanYou** [comment]
Ok, thanks

**Comment by MoreRobustThanYou** [comment]
OK, wew... I made plasma-glass an alloy (smelted in the ORM/smelter) and added stupid quotes to armor

**Comment by MoreRobustThanYou** [comment]
@Cyberboss Anything else to fix?

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 26]
```diff
@@ -55,8 +55,8 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 			RG.add_fingerprint(user)
 			var/replace = user.get_inactive_held_item()==src
 			V.use(1)
-			G.use(1)
-			if(QDELETED(G) && replace)
+			src.use(1)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: no src here

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 27]
```diff
@@ -99,8 +99,8 @@ GLOBAL_LIST_INIT(pglass_recipes, list ( \
...
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: no src here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MoreRobustThanYou** [comment]
Done... I think?

**Comment by MoreRobustThanYou** [comment]
SHOULD be merge-ready now. Fixed a small bug that caused plasma glass to appear as normal glass... but should work. Not sure

**Comment by MoreRobustThanYou** [comment]
no longer WIP

**Comment by MoreRobustThanYou** [comment]
And IT. IS. DONE!

**Comment by MoreRobustThanYou** [comment]
@Cyberboss I already did your changes, I think

**Comment by Tokiko1** [comment]
If you want to be able to put those windows into high temperature areas like burn chambers and not have them break immediatly, you'll probably want to go much higher than just 2500/4000 in heat_resistance. Try something like 25.000 for regular and 50.000 for reinforced windows instead.

**Comment by MoreRobustThanYou** [comment]
@Tokiko1 done

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 28]
```diff
@@ -50,15 +50,13 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+			var/obj/item/stack/sheet/rglass/RG = new (get_turf(user))
 			RG.add_fingerprint(user)
-			var/obj/item/stack/sheet/glass/G = src
-			src = null
-			var/replace = (user.get_inactive_held_item()==G)
+			var/replace = user.get_inactive_held_item()==src
 			V.use(1)
-			G.use(1)
-			if (!G && replace)
+			src.use(1)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: remove src

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 29]
```diff
@@ -67,6 +65,51 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
...
+	add_fingerprint(user)
+
+	if(istype(W, /obj/item/stack/rods))
+		var/obj/item/stack/rods/V = W
+		if (V.get_amount() >= 1 && get_amount() >= 1)
+			var/obj/item/stack/sheet/rglass/plasma/RG = new (get_turf(user))
+			RG.add_fingerprint(user)
+			var/replace = user.get_inactive_held_item()==src
+			V.use(1)
+			src.use(1)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: remove src

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 30]
```diff
@@ -85,10 +128,14 @@ GLOBAL_LIST_INIT(reinforced_glass_recipes, list ( \
...
 	materials = list(MAT_METAL=MINERAL_MATERIAL_AMOUNT/2, MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
 	origin_tech = "materials=2"
-	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 70, acid = 100)
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 70, "acid" = 100)
 	resistance_flags = ACID_PROOF
 	merge_type = /obj/item/stack/sheet/rglass
 
+/obj/item/stack/sheet/rglass/attackby(obj/item/W, mob/user, params)
+	..()
+	add_fingerprint(user)
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/glass.dm]: add_fingerprint before ..()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MoreRobustThanYou** [comment]
@Cyberboss  all done

**Review by Cyberboss — APPROVED** [review]

**Comment by MoreRobustThanYou** [comment]
@Cyberboss a pic has been added to the post

**Comment by MoreRobustThanYou** [comment]
Yay, time to add this to the maps, now.

**Pull request merged**